### ISSUE TITLE
Horizontal rule identified in code block

### ISSUE THREAD

**Issue by treyhunner** [rvben/rumdl#43]
This file:

    ---
    title: Python's many command-line utilities
    description: Every command-line tool included with Python. These can be run with `python -m module_name`.
    ---

    # `runpy`

    The `runpy` module is used for dynamically running a given Python module by its name.
    The fact that it has a command-line interface is a bit odd, since it essentially does what Python already does for us!

    Here's `runpy` running `runpy` running `runpy` running the `unittest` module:

    ```bash
    $ python -m runpy runpy runpy unittest

    ----------------------------------------------------------------------
    Ran 0 tests in 0.000s

    NO TESTS RAN
    ```

Shows this error:

```bash
$ uv tool run rumdl check --disable MD013 file14.md
file14.md:16:1: [MD035] Horizontal rule style should be "---" [*]

Issues: Found 1 issues in 1 file (23ms)
Run with `--fix` to automatically fix 1 of the 1 issues
```

**Comment by rvben** [comment]
✅ **Fixed in v0.0.113**

MD035 now properly respects code block boundaries and won't flag horizontal rule patterns inside code blocks. The fix uses the existing  infrastructure to skip code content.

Thanks for the detailed report\!
